src/sentry/workflow_engine/models/action.py

```diff
@@ -110,8 +110,8 @@ def trigger(self, event_data: WorkflowEventData, detector: Detector) -> None:
             },
         )
 
-    def get_dedup_key(self) -> str:
-        key_parts = [self.type]
+    def get_dedup_key(self, workflow_id: int) -> str:
+        key_parts = [self.type, str(workflow_id)]
```

Comment:
nit: can we make the workflow id `int | None = None` and update the keyparts to check if there's an ID or not? my thought is, if i just have a list of actions for a single workflow, then i don't need to worry about the workflow_id.

that might help make this method a bit more reusable for other cases where an `id` isn't a great way to determine uniqueness.